stl/inc/format

```diff
@@ -2497,11 +2497,13 @@ _NODISCARD _OutputIt _Fmt_write(
     // Add 3 to the bit width so we always round up on the division.
     // Divide that by the amount of bits a hexit represents (log2(16) = log2(2^4) = 4).
     // Add 2 for the 0x prefix.
-    auto _Width = 2 + static_cast<int>(_STD bit_width(reinterpret_cast<uintptr_t>(_Value)) + 3) / 4;
+    int _Width;
 
     // Since the bit width of 0 is 0, special case it instead of complicating the math even more.
     if (_Value == nullptr) {
         _Width = 3;
+    } else {
+        _Width = 2 + static_cast<int>(_STD bit_width(reinterpret_cast<uintptr_t>(_Value)) + 3) / 4;
     }

```

Comment:
It seems more natural to just:
`const auto _Width = _Value == nullptr ? 3 : 2 + static_cast<int>(_STD bit_width(reinterpret_cast<uintptr_t>(_Value)) + 3) / 4;`

Reply:
I personally prefer how the author had it before because there isn't so much on a single line. Let's see what the maintainers prefer.

Reply:
We have a general preference to avoid declaring variables without initializers, and a mild inclination to avoid the conditional operator since it tends to result in overly nested expressions. (I think @barcharcraz has much stronger personal feelings about the conditional operator.) So I'd probably write this as:
```
int _Width = 3;

if (_Value != nullptr) {
    _Width = /* blah blah */;
}
```

Reply:
One tiny difference is you lose `const`ness in such cases.

Reply:
One tiny difference is you lose `const`ness in such cases.

Reply:
Pre-existing: it's a bit weird that the `static_cast` is applied to the addition instead of only to the result of `bit_width`.